File: problems/0766.0_Toeplitz_Matrix.py
'''
Runtime: 92 ms, faster than 91.65% of Python3 online submissions for Toeplitz Matrix.
Memory Usage: 13.9 MB, less than 36.74% of Python3 online submissions for Toeplitz Matrix.
'''
class Solution:
    def isToeplitzMatrix(self, matrix: List[List[int]]) -> bool:
        m, n = len(matrix), len(matrix[0])
        for r in range(1, m):
            for c in range(1, n):
                if matrix[r][c] != matrix[r - 1][c - 1]:
                    return False
        return True
        
        

File: csharp/0766.0_Toeplitz_Matrix.cs
/*
Runtime: 144 ms, faster than 74.77% of C# online submissions for Toeplitz Matrix.
Memory Usage: 41.7 MB, less than 23.36% of C# online submissions for Toeplitz Matrix.
*/

public class Solution {
    public bool IsToeplitzMatrix(int[][] matrix) {
        int rowLen = matrix.Length;
        int colLen = matrix[0].Length;
        for (int i = 0; i < rowLen - 1; i++) {
            for (int j = 0; j < colLen - 1; j++) {
                if (matrix[i][j] != matrix[i + 1][j + 1]) return false;
            }
        }
        return true;
    }
}
